### PR TITLE
Add content length on pendant request + Override Data

### PR DESCRIPTION
**Pull request by manoukianv**
This PR is to update de pendant request to add content-size in header to optimize http request and add override data.

This will be helpfull in IOT (arduino / stm32 / etc) project to reduce to get the data : when there is no size content, the iot frameworks wait 1s before get the data.

I update Pendant tools to build a bCNC panel (hardware).

### CODE REVIEW

File: bCNC/Pendant.py [hunk 1]
```diff
@@ -32,7 +32,7 @@
 	Image = None
 
 try:
-	import BaseHTTPServer as HTTPServer
+		import BaseHTTPServer as HTTPServer
```
Comment (Harvie) [inline bCNC/Pendant.py]: This indentation seems to be redundant to me...

File: bCNC/Pendant.py [hunk 2]
```diff
@@ -89,26 +91,28 @@ def do_GET(self):
...
 
 		elif page == "/state":
-			self.do_HEAD(200, content="text/text")
 			tmp = {}
-			for name in ["controller", "state", "pins", "color", "msg", "wx", "wy", "wz", "G"]:
+			for name in ["controller", "state", "pins", "color", "msg", "wx", "wy", "wz", "G", "OvFeed", "OvRapid", "OvSpindle"]:
 				tmp[name] = CNC.vars[name]
+			contentToSend = json.dumps(tmp)
+			self.do_HEAD(200, content="text/text", cl=len(contentToSend))
 			self.wfile.write(json.dumps(tmp))
```
Comment (Harvie) [inline bCNC/Pendant.py]: probably no need to call json.dumps() again, perhaps we can reuse contentToSend?

**Comment by Harvie** [comment]
Thank you! Seems sane, i will merge this. Can you please fix the cosmetic problems i've described?

File: bCNC/Pendant.py (hunk 1 above)
Comment (manoukianv) [reply bCNC/Pendant.py]: You right, i will fix it

File: bCNC/Pendant.py (hunk 2 above)
Comment (manoukianv) [reply bCNC/Pendant.py]: Yes, i fix it, it's not optimized.

**Comment by Harvie** [comment]
Thanks. Please note that if you are developing some improved web pendant, there are already people working on this, so be sure to join efforts with them: #1121

**Pull request merged**